bedwars-plugin/src/main/java/com/andrei1058/bedwars/arena/tasks/EggBridgeTask.java

```diff
@@ -75,54 +76,40 @@ public IArena getArena() {
 
     @Override
     public void run() {
-
         Location loc = getProjectile().getLocation();
+        int maxBuildY = getArena().getConfig().getInt(ConfigPath.ARENA_CONFIGURATION_MAX_BUILD_Y);
 
-        if (getProjectile().isDead()
-                || !arena.isPlayer(getPlayer())
-                || getPlayer().getLocation().distance(getProjectile().getLocation()) > 27
-                || getPlayer().getLocation().getY() - getProjectile().getLocation().getY() > 9) {
+        if (loc.getBlockY() >= maxBuildY) {
```

Comment:
This is already verified (or it should) via Misc#isBuildProtected. By using it here you are blocking it, but there are 3 possible bulding locations down in the code, so I'd say to remove this control.